go/src/koding/klient/machine/dynamic.go

```diff
@@ -192,10 +192,14 @@ func (dc *DynamicClient) cron() {
 
 // tryUpdate uses client builder to ping the machine and updates dynamic client
 // if machine address changes.
-func (dc *DynamicClient) tryUpdate(addr *Addr) {
+func (dc *DynamicClient) tryUpdate(addr *Addr, islog *bool) {
 	stat, a, err := dc.opts.Builder.Ping(dc.opts.AddrFunc)
 	if err != nil {
-		dc.log.Warning("Machine ping error: %s", err)
+		if *islog {
```

Comment:
Just to note that, the code you pointed is a one-off script which is run manually. It is not deployed, it is not monitored, it is not in any critical path of our production workload or any user facing features/infrastructure. Please check tunnelproxymanager as an example that contains many pointer variables because of aws-sdk-go's structs.

It is always better to be constructive than destructive. Pointing fingers is not from the constructive ones.

The code you linked also irrelevant to your proposal. It is an exposed function.